lua/telescope/actions/history.lua

```diff
@@ -0,0 +1,218 @@
+    obj.limit = conf.history_limit
+  end
+
+  obj.path = vim.fn.expand(conf.history_location)
+  local p = Path:new(obj.path)
+  if not p:exists() then p:touch({ parents = true }) end
+
+  obj.content = Path:new(obj.path):readlines()
+  obj.index = #obj.content
+  table.remove(obj.content, obj.index)
+
+  return setmetatable(obj, self)
+end
+
+function SimpleHistory:reset()
+  if not self.enabled then return end
+  self.index = #self.content + 1
+end
+
+function SimpleHistory:append(line)
+  if not self.enabled then return end
+
+  if line ~= '' then
+    if self.content[#self.content] ~= line then
+      table.insert(self.content, line)
+
+      local len = #self.content
+      if self.limit and len > self.limit then
+        -- We need to remove the first diff items not only the first one
+        -- because a history can already be like 60 items but the user changes it
+        -- to 50. Now we need to remove the no longer needed items.
+        -- We could do this check on startup as well but currently i don't think its
+        -- necessary.
+        local diff = len - self.limit
+        for i = diff, 1, -1 do
+          table.remove(self.content, i)
+        end
+        write_async(self.path, table.concat(self.content, '\n') .. '\n', 'w')
+      else
+        append_async(self.path, line .. '\n')
+      end
+    end
+  end
+  self:reset()
+end
+
+function SimpleHistory:get_next()
+  if not self.enabled then return end
+
+  local next_idx = self.index + 1
+  if next_idx <= #self.content then
+    self.index = next_idx
+    return self.content[next_idx]
+  end
+  self.index = #self.content + 1
+  return nil
+end
+
+function SimpleHistory:get_prev()
+  if not self.enabled then return end
+
+  local next_idx = self.index - 1
+  if next_idx >= 1 then
+    self.index = next_idx
+    return self.content[next_idx]
+  end
+  return nil
+end
+
+m.get_simple_history = function(...)
+  return SimpleHistory:new(...)
+end
+
+------------------------------------------------------------------------------
+------------------------------------------------------------------------------
+-- DB HISTORY SCRATCH. MOVING TO EXTENSION AFTER MAKING IT WORK --------------
+
+local sql = require('sql')
+
+-- TODO(conni2461): Serialize metadata
+-- TODO(conni2461): CURRENTLY HARDCODED picker and cwd
+local SmartHistory = {}
+SmartHistory.__index = SmartHistory
+
+function SmartHistory:new()
+  local obj = {
+    enabled = true
+  }
+  if not conf.history_location then
+    obj.enabled = false
+    return setmetatable(obj, self)
+  end
+  if conf.history_limit then
+    obj.limit = conf.history_limit
+  end
+
+  obj.path = vim.fn.expand(conf.history_location)
+  obj.db = sql:open(obj.path)
+  -- any chance i can get a table object here?!?
+  -- because using obj.db:table to create one feels weird because i can't pass in
+  -- a schema at the same time. WTF tami am i just stupid.
+  -- I think the readme is confusing LULW
+  obj.db:create("histories", {
```

Comment:
```
obj.data = db:table("history",{nocache = true, schema = {..., ensure = true, id=true}
```

This a better approach so that later we can do obj.data:insert(), obj.data:get() .. etc and with sql.new we don't need to worry about opening or closing connections.

id=true is as same as how you defined the id above. nocache=true because history will be always updated on every run.